tedana/selection/select_comps.py

```diff
@@ -284,90 +281,97 @@ def selcomps(seldict, comptable, mmix, manacc, n_echos):
+        candartA = (candartA[comptable.loc[candartA, 'variance explained'] >
+                    varex_upper * EXTEND_FACTOR])
         comptable.loc[candartA, 'classification'] = 'rejected'
         comptable.loc[candartA, 'rationale'] += 'I009;'
         midk = np.union1d(midk, candartA)
+        unclf = np.setdiff1d(unclf, midk)
 
         # Rejection candidate based on artifact type B: candartB
-        candartB = comptable.loc[acc].loc[
-            comptable.loc[acc, 'd_table_score_scrub'] >
-            num_acc_guess * n_decision_metrics * HIGH_PERC / 100.].index.values
-        candartB = np.intersect1d(
-            candartB,
-            candartB[comptable.loc[candartB, 'variance explained'] > varex_lower * EXTEND_FACTOR])
+        conservative_guess2 = num_acc_guess * HIGH_PERC / 100.
+        candartB = unclf[comptable.loc[unclf, 'd_table_score_scrub'] > conservative_guess2]
+        candartB = (candartB[comptable.loc[candartB, 'variance explained'] >
+                    varex_lower * EXTEND_FACTOR])
         comptable.loc[candartB, 'classification'] = 'rejected'
         comptable.loc[candartB, 'rationale'] += 'I010;'
         midk = np.union1d(midk, candartB)
+        unclf = np.setdiff1d(unclf, midk)
 
         # Find components to ignore
+        # Ignore high variance explained, poor decision tree scored components
         new_varex_lower = stats.scoreatpercentile(
-            comptable.loc[acc[:num_acc_guess], 'variance explained'],
+            comptable.loc[unclf[:num_acc_guess], 'variance explained'],
             LOW_PERC)
-        candart = comptable.loc[acc].loc[
-            comptable.loc[acc, 'd_table_score_scrub'] >
-            num_acc_guess * n_decision_metrics].index.values
-        ign_add0 = np.intersect1d(
-            candart[comptable.loc[candart, 'variance explained'] > new_varex_lower],
-            candart)
+        candart = unclf[comptable.loc[unclf, 'd_table_score_scrub'] > num_acc_guess]
+        ign_add0 = candart[comptable.loc[candart, 'variance explained'] > new_varex_lower]
         ign_add0 = np.setdiff1d(ign_add0, midk)
         comptable.loc[ign_add0, 'classification'] = 'ignored'
         comptable.loc[ign_add0, 'rationale'] += 'I011;'
         ign = np.union1d(ign, ign_add0)
+        unclf = np.setdiff1d(unclf, ign)
 
+        # Ignore low Kappa, high variance explained components
         ign_add1 = np.intersect1d(
-            acc[comptable.loc[acc, 'kappa'] <= kappa_elbow],
-            acc[comptable.loc[acc, 'variance explained'] > new_varex_lower])
+            unclf[comptable.loc[unclf, 'kappa'] <= kappa_elbow],
+            unclf[comptable.loc[unclf, 'variance explained'] > new_varex_lower])
         ign_add1 = np.setdiff1d(ign_add1, midk)
         comptable.loc[ign_add1, 'classification'] = 'ignored'
         comptable.loc[ign_add1, 'rationale'] += 'I012;'
 
+    # at this point, unclf is equivalent to accepted
```

Comment:
Not sure I understand having this comment here, but happy to defer to you ! Just pointing it out as potentially confusing.

Reply:
`unclf` isn't used to define accepted components. All components are initially classified as "accepted" and then the function builds up the ignored and rejected classifications. The remaining list of "unclassified" components ends up being the list of components remaining with the "accepted" classification in the component table. I'm not sure how best to convey that.

Reply:
I think I was just confused with the comment being there since `unclf` is not actually called again at this point. I get what you're trying to say, but my first reaction was sort of like "why is it here?" One option would be to do something like `del unclf`, though I'm not sure that actually clarifies anything.

Very light opinion, so feel free to just ignore this point entirely !